power_grid_model_c/power_grid_model/include/power_grid_model/main_core/update.hpp

```diff
@@ -96,22 +101,131 @@ inline void update_inverse(MainModelState<ComponentContainer> const& state, Forw
-    y_bus.update_admittance(math_model_param);
+inline void update_y_bus_increment(YBus<sym>& y_bus, std::shared_ptr<MathModelParam<sym> const> const& math_model_param,
+                                   bool increment) {
+    auto branch_param_to_change_views =
+        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered([&math_model_param](Idx i) {
+            return math_model_param->branch_param[i].yff() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].yft() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytf() != ComplexTensor<sym>{0.0} ||
+                   math_model_param->branch_param[i].ytt() != ComplexTensor<sym>{0.0};
+        });
+    auto shunt_param_to_change_views =
+        boost::irange(0, math_model_param->branch_param.size()) | boost::adaptors::filtered([&math_model_param](Idx i) {
+            return math_model_param->shunt_param[i] != ComplexTensor<sym>{0.0};
+        });
+
+    MathModelParamIncrement<sym> math_model_param_incrmt;
+    math_model_param_incrmt.branch_param_to_change = {branch_param_to_change_views.begin(),
+                                                      branch_param_to_change_views.end()};
+    math_model_param_incrmt.shunt_param_to_change = {shunt_param_to_change_views.begin(),
+                                                     shunt_param_to_change_views.end()};
+
+    auto param_incrmt_ptr = std::make_shared<MathModelParamIncrement<sym> const>(math_model_param_incrmt);
+
+    y_bus.update_admittance_increment(math_model_param, param_incrmt_ptr,
+                                      !increment); /* param, changed_param, is_decrement */
 }
 
 template <bool sym>
 inline void update_y_bus(MathState& math_state, std::vector<MathModelParam<sym>> const& math_model_params,
-                         Idx n_math_solvers) {
+                         Idx n_math_solvers, std::vector<std::vector<Idx2D>> const& seq_idx_map) {
     for (Idx i = 0; i != n_math_solvers; ++i) {
+        auto seq_idx_map_copy = seq_idx_map;
         if constexpr (sym) {
             update_y_bus(math_state.y_bus_vec_sym[i],
-                         std::make_shared<MathModelParam<sym> const>(std::move(math_model_params[i])));
+                         std::make_shared<MathModelParam<sym> const>(std::move(math_model_params[i])),
+                         std::make_shared<std::vector<std::vector<Idx2D>> const>(std::move(seq_idx_map_copy)));
 
         } else {
             update_y_bus(math_state.y_bus_vec_asym[i],
-                         std::make_shared<MathModelParam<sym> const>(std::move(math_model_params[i])));
+                         std::make_shared<MathModelParam<sym> const>(std::move(math_model_params[i])),
+                         std::make_shared<std::vector<std::vector<Idx2D>> const>(std::move(seq_idx_map_copy)));
         }
     }
```

Comment:
you can also do something like

```cpp
    auto const& ybus_vec = [&math_state] -> auto const& {
        if constexpr (sym) {
            return math_state.y_bus_vec_sym;
        } else {
            return math_state.y_bus_vec_asym;
        }();

        for (Idx i = 0; i != n_math_solvers; ++i) {
            auto seq_idx_map_copy = seq_idx_map;
            update_y_bus(math_state.y_bus_vec_asym[i],
                         std::make_shared<MathModelParam<sym> const>(std::move(math_model_params[i])),
                         std::make_shared<std::vector<std::vector<Idx2D>> const>(std::move(seq_idx_map_copy)));
        }
    }

Reply:
Not sure if correct... Apart from reducing duplicates, what's the benefit here?

Reply:
Resolved in latest commit.